### ISSUE TITLE
[New Analyzer Suggestion] Remove Unnecessary ByVal

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#3690]
The `ByVal` modifier is the default and unnecessary to include.

A new analyzer to detect the usages of `ByVal` in Visual Basic and a CodeFix to remove it will help cleaning the code and make it less noisy.

Example:

Given the following code:

```vb
Public Sub SayHello(ByVal name As String)
    Console.WriteLine($"Hello, {name}!")
End Sub
```

The analyzer should report unnecessary `ByVal` and the code fix turns the code to:

```vb
Public Sub SayHello(name As String)
    Console.WriteLine($"Hello, {name}!")
End Sub
```

The removal of ByVal, as mentioned before, makes the code much less noiser.
It also helps identifying ByRef easily. For example, it's easy to spot that `param3` is `ByRef` in the following snippet:

```vb
Public Sub DoSomething(param1, param2, ByRef param3, param4, param5)
    ' Do something.
End Sub
```

But it's harder to spot that in the following snippet.

```vb
Public Sub DoSomething(ByVal param1, ByVal param2, ByRef param3, ByVal param4, ByVal param5)
    ' Do something.
End Sub
```

**Comment by Youssef1313** [comment]
If this suggestion is accepted, I'd like to try implementing it. Is there a guide on that?

**Comment by mavasani** [comment]
This is more of a code style analyzer, that belongs to Roslyn repo. This repo targets semantic/code quality analyzers. I will move it over.
